src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityWorldAccelerator.java

```diff
@@ -139,72 +105,102 @@ public void addToolUsages(ItemStack stack, @Nullable World world, List<String> t
-                            ITickable neighbourTickTile = (ITickable) neighbourTile;
-                            for (int i = 0; i < speed; i++) {
-                                neighbourTickTile.update();
-                            }
-                        }
-                    }
-                } else {
-                    energyContainer.removeEnergy(energyPerTick / 2);
-                    if (range == null) {
-                        int area = getTier() * 2;
-                        range = () -> BlockPos.getAllInBoxMutable(currentPos.add(-area, -area, -area), currentPos.add(area, area, area));
-                    }
-                    for (BlockPos.MutableBlockPos pos : range.get()) {
-                        if (pos.getY() > 256 || pos.getY() < 0) { // Early termination
-                            continue;
-                        }
-                        if (world.isBlockLoaded(pos)) {
-                            for (int i = 0; i < speed; i++) {
-                                if (GTValues.RNG.nextInt(100) < getTier()) {
-                                    // Rongmario:
-                                    // randomTick instead of updateTick since some modders can mistake where to put their code.
-                                    // Fresh IBlockState before every randomTick, this could easily change after every randomTick call
-                                    IBlockState state = world.getBlockState(pos);
-                                    Block block = state.getBlock();
-                                    if (block.getTickRandomly()) {
-                                        block.randomTick(world, pos.toImmutable(), state, world.rand);
-                                    }
-                                }
-                            }
-                        }
-                    }
-                }
+        }
+        return true;
+    }
+
+    private boolean handleRandomTickMode() {
+        long energyUsage = getRandomTickModeAmperage() * GTValues.V[getTier()];
+        if (energyContainer.getEnergyStored() < energyUsage) return false;
+        energyContainer.removeEnergy(energyUsage);
+
+        World world = getWorld();
+        // room for hitting ourselves randomly, or blocks not loaded, or blocks outside of height limits
+        int attempts = successLimit * 3;
+        BlockPos cornerPos = getCornerPos();
+        BlockPos.MutableBlockPos mutablePos = new BlockPos.MutableBlockPos(cornerPos);
+        for (int i = 0, j = 0; i < successLimit && j < attempts; j++) {
+            int x = GTValues.RNG.nextInt(getTier() - 1);
+            int y = GTValues.RNG.nextInt(getTier() - 1);
+            int z = GTValues.RNG.nextInt(getTier() - 1);
+            mutablePos.setPos(
+                    cornerPos.getX() + x,
+                    cornerPos.getY() + y,
+                    cornerPos.getZ() + z);
+
+            if (cornerPos.getY() > 256 || cornerPos.getY() < 0) continue;
```

Comment:
We can try to use the world's max height with a method in the `World` class outside this loop, and use it here.

Is this preferable?

Also I think `cornerPos` should be `mutablePos` here.